### PR TITLE
 Create logger hierarchy for config and incoming / outgoing loggers

### PR DESCRIPTION
**Pull request by danielbotros**
### Description
RSDK-8084: Add config sublogger for all config related log lines
* This sublogger will allow users to configure log lines around config processing and watching. 

RSDK-8091: Create and organize outgoing network logger and subloggers 
* This sublogger will allow users to configure log lines related to outgoing network events, such as dialing GRPC connections

RSDK-8092: Create and organize incoming network logger and subloggers 
* This sublogger will allow users to configure log lines related to incoming network events, such as sessions and signaling. NOTE: There exists no loggers for grpc_requests, so that node currently does not exist.

### CODE REVIEW

File: grpc/shared_conn.go
Comment (danielbotros) [inline grpc/shared_conn.go:159]: From what I understand, this is where `sc.logger` gets initialized. I think it is valuable to keep using the `moduleLogger` for the module logger name prefix, but since this logger deals with outgoing network connections, we want to be able to configure this using the `network_outgoing` string as well. This deviates from the `logging.GetOrNewLogger("rdk.network_outgoing")` pattern slightly in order to keep the module name and still be configurable using e.g. `*.network_outgoing.*`. I am open to suggestions or reverting this change.

File: logging/net_appender.go
Comment (danielbotros) [inline logging/net_appender.go:442]: This logger is guaranteed to exist from the line above, but this was the most straightforward way to retrieve it.

File: logging/logger_registry.go
Comment (danielbotros) [inline logging/logger_registry.go:158]: I don't think there is a clear natural place to initialize a e.g. `rdk.network_outgoing` logger so this helper lets us safely retrieve and use the same logger object globally without fear of overwriting a map entry to a new logger with the same name.

File: module/module.go
Comment (danielbotros) [inline module/module.go:239]: some of these are a bit dubious as to whether or not they are incoming or outgoing network events in my opinion

File: grpc/shared_conn.go
Comment (benjirewis) [reply grpc/shared_conn.go:159]: Thanks for calling this out. So, logs from the shared conn to a module from the module manager will have the logger name: `rdk.modmanager.[modulename].network_outgoing.conn`? I think that's fine. I can't really think of a better option: `rdk.network_outgoing.modmanager.module.conn` would _also_ be slightly inconvenient, as if a user tried to set `rdk.modmanager.*`, they'd miss this logger. Also tagging @dgottlieb for visibility on our changes here.

File: logging/logger_registry.go
Comment (benjirewis) [inline logging/logger_registry.go:153]: Could we leave a `TODO` comment to consider removing this with https://viam.atlassian.net/browse/RSDK-8291? See my other comment on why that ticket is relevant.

File: module/module.go
Comment (benjirewis) [reply module/module.go:239]: Why would we put this logger under `rdk.network_outgoing`? From what I can tell it's a logger used as part of connection establishment from a Go module back to the RDK. I think we should continue to use `logger`; that's the module's author's logger.

File: robot/client/client.go
Comment (benjirewis) [inline robot/client/client.go:348]: This is a tough one, and I think it's slightly incorrect. You're making the gRPC dial logger `rdk.network_outgoing` for all Go SDK clients with this change. While the RDK uses Go SDK clients to connect to remotes, we don't want to mess with the Go SDK for people that use it as an actual SDK. I think you'll want to change [this](https://github.com/viamrobotics/rdk/blob/main/robot/impl/local_robot.go#L1129) logger instead.

**Review by benjirewis — CHANGES_REQUESTED** [review]
Thanks for combining these smaller changes into one PR! Some requests here.

File: grpc/shared_conn.go
Comment (danielbotros) [reply grpc/shared_conn.go:159]: correct, I also agree this is probably the best we can do for now

File: module/module.go
Comment (danielbotros) [reply module/module.go:239]: discussed off PR, we are changing all `network_outgoing` / `network_incoming` to `networking`

File: robot/client/client.go
Comment (danielbotros) [reply robot/client/client.go:348]: ah yeah that makes sense, thanks for calling out

Comment (danielbotros) [reply robot/client/client.go]: I decided to not go ahead with that change and instead change `logging.GetOrNewLogger("rdk.networking")` to `rc.logger.Sublogger("networking")` so we still get the `networking` suffix (and so will SDK users, but I think this is ok). The line you linked would only change connection logging for remotes, where I think it that case it would be valuable to still keep the resource name as the logger name (as well as adding the `networking` suffix).

**Review by benjirewis — APPROVED** [review]
Approving so I don't block this PR while I'm OOO.

**Review by puranamp — APPROVED** [review]

**Review by maximpertsov — APPROVED** [review]

**Pull request merged**
